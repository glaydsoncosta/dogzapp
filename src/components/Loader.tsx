import React from 'react';
import { ActivityIndicator, StyleSheet, Text, View } from 'react-native';
import { primary } from '../styles/colors';

type LoaderProps = {
  loadingText?: string;
};

export default function Loader(props: LoaderProps) {
  const { loadingText } = props;
  return (
    <View style={styles.mainContainer}>
      <ActivityIndicator color={primary} size="small" />
      <Text style={styles.loadingTextStyle}>{loadingText}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  loadingTextStyle: {
    marginTop: 10
  }
});
