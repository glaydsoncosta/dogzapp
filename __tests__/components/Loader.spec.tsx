import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../../src/components/Loader';

test('Loader snapshot', () => {
  const snapShot = renderer.create(<Loader />).toJSON();
  expect(snapShot).toMatchSnapshot();
});
