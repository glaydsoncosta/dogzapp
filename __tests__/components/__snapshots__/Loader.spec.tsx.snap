// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#673AB7"
    size="small"
  />
  <Text
    style={
      {
        "marginTop": 10,
      }
    }
  />
</View>
`;
